src/NetTopologySuite.IO.ShapeFile/Streams/ManagedStreamProvider.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace NetTopologySuite.IO.Streams
+{
+    /// <summary>
+    /// A stream provider that provides <see cref="Stream"/>s who are managed by the application.
+    /// </summary>
+    public class ManagedStreamProvider : IStreamProvider
```

Comment:
Name suggestion: perhaps `ExternallyManagedStreamProvider`?

Reply:
I Can live with that